anndata/tests/test_awkward.py

```diff
@@ -97,6 +98,21 @@ def _assign():
         _assign()
 
 
+@pytest.mark.parametrize("key", ["obsm", "varm", "uns"])
+@pytest.mark.skipif(
+    awkward_version >= 2, reason="This test is only applies for awkward versions 1.x"
+)
+def test_no_awkward_v1(key):
+    """Assigning an awkward v1 array to anndata should fail"""
+    import awkward as akv1
+
+    v1arr = akv1.Array([1, 2, 3, 4])
+
+    adata = AnnData(np.ones((4, 4)))
+    with pytest.raises(AttributeError):
+        getattr(adata, key)["test"] = v1arr
```

Comment:
1. You can't assign a v1 array to `obsm`/`varm`, because the `dim_len` doesn't dispatch on the awkward handler and raises an error. However the error message doesn't tell anything about the `v1` array being the problem. Is this good enough? 
1. You *can* assign a v1 array to `uns`, because there is no check whatsoever (-> this test fails!). I'm not even sure where I would add one, as `.uns` can be just a regular dictionary? 

@ivirshup, how shall we go about this? Maybe just document it properly? We will anyway require the `v2` version as soon as it is released.

Reply:
Yeah, I think just documenting properly would be fine. We could make a mapping type that converts, but I think we're fine without.

Reply:
yeah, that would just mean adding code that will be obsolete in a few months. 
Do you want me to remove this test altogether or keep it for `obsm` and `varm`?